snapshot/generate.go

```diff
@@ -615,31 +616,27 @@ func (dl *diskLayer) generate(stats *generatorStats) {
 			snapAccountWriteCounter.Inc(time.Since(start).Nanoseconds())
 			return nil
 		}
-		// Retrieve the current account and flatten it into the internal format
-		var acc struct {
-			Nonce    uint64
-			Balance  *big.Int
-			Root     common.Hash
-			CodeHash []byte
-		}
-		if err := rlp.DecodeBytes(val, &acc); err != nil {
+		serializer := account.NewAccountSerializer()
+		if err := rlp.DecodeBytes(val, serializer); err != nil {
 			logger.Crit("Invalid account encountered during snapshot creation", "err", err)
 		}
+		acc := serializer.GetAccount()
 		// If the account is not yet in-progress, write it out
 		if accMarker == nil || !bytes.Equal(accountHash[:], accMarker) {
 			dataLen := len(val) // Approximate size, saves us a round of RLP-encoding
+			// TODO-Klaytn-Snapshot consider slim account in order to save space for emptyRootHash, emptyCodeHash
```

Comment:
@jeongkyun-oh Do we need something more to implement here?

Reply:
@kjhman21 The original implementation modified their account model to reduce data size. So, we could do that also but it is optional so I leave a comment. It is fine as it is, though.

Reply:
@jeongkyun-oh Can we do more? I think we do not have emptyRootHash or codeHash for EOAs.

Reply:
@kjhman21 How about SmartContractAccount?